src/NetscriptFunctions.ts

```diff
@@ -1414,34 +1405,30 @@ export const ns: InternalAPI<NSFull> = {
     return status.res;
   },
   scriptRunning: (ctx) => (_scriptname, _hostname) => {
-    const scriptname = helpers.string(ctx, "scriptname", _scriptname);
+    const scriptname = helpers.scriptPath(ctx, "scriptname", _scriptname);
     const hostname = helpers.string(ctx, "hostname", _hostname);
     const server = helpers.getServer(ctx, hostname);
-    for (let i = 0; i < server.runningScripts.length; ++i) {
-      if (server.runningScripts[i].filename == scriptname) {
-        return true;
-      }
-    }
-    return false;
+    return server.isRunning(scriptname);
   },
   scriptKill: (ctx) => (_scriptname, _hostname) => {
-    const scriptname = helpers.string(ctx, "scriptname", _scriptname);
+    const path = helpers.scriptPath(ctx, "scriptname", _scriptname);
     const hostname = helpers.string(ctx, "hostname", _hostname);
     const server = helpers.getServer(ctx, hostname);
     let suc = false;
-    for (let i = 0; i < server.runningScripts.length; i++) {
-      if (server.runningScripts[i].filename == scriptname) {
-        killWorkerScript({ runningScript: server.runningScripts[i], hostname: server.hostname });
-        suc = true;
-        i--;
+
+    const pattern = matchScriptPathExact(escapeRegExp(path));
```

Comment:
I struggled (and failed) to come up with a good name here. Do you have a suggestion?

Reply:
I would keep the name the same but rewrite it to actually take in a ScriptFilePath directly, do the escaping, and return the regex, instead of expecting the input to be already-escaped. Then the name would make sense with the function, and it would also be a far more usable function (would not be required to import a regex escaper every time you want to use one of these functions).

Same with the unanchored one - it should take in the filter pattern, then escape it, and then return an appropriate regex. This does mean the unanchored one would have to separately return a regex instead of wrapping the exact function.

Reply:
I can make the ExactPath version take a FilePath, but the unanchored one can't, since at least some of its callers provide a pattern as opposed to a path. So then there's that weird dichotomy.

Reply:
Yeah it's just a bit cumbersome to use the functions right now. But not a big deal, I'm just going to merge this in so we can get more playtesting time in on this before 2.3 goes live.

Reply:
Makes sense, if you can think of better naming/a less awkward way to arrange this later, I'm all for it. It's mostly just `--grep` in ps that's messing things up.